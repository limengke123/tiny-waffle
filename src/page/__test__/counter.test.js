import React from 'react'
import { render, mount, shallow } from 'enzyme'
import Counter from '../counter'

describe('Count', () => {
    const randomSpy = jest.spyOn(Math, 'random').mockImplementation(() => 0.5)

    afterAll(() => {
        randomSpy.mockRestore()
    })

    it('should render correctly', () => {
        const wrapper = render(<Counter />)
        expect(wrapper).toMatchSnapshot()
    })

    it('should handle change step correctly', () => {
        const wrapper = shallow(<Counter />)
        const instance = wrapper.instance()
        instance.changeStep()
        expect(instance.state).toMatchObject({
            count: {
                modifyInfo: {
                    step: 6
                }
            }
        })
    })

    it('should handle increase correctly', () => {
        const wrapper = shallow(<Counter />)
        const instance = wrapper.instance()
        instance.increase()
        expect(instance.state).toMatchObject({
            count: {
                number: 1,
                modifyInfo: {
                    type: 'Addition'
                }
            }
        })

        instance.state.count.modifyInfo.step = 3
        instance.state.count.number = 4
        instance.increase()
        expect(instance.state).toMatchObject({
            count: {
                number: 7,
                modifyInfo: {
                    type: 'Addition'
                }
            }
        })
    })

    it('should handle decrease correctly', () => {
        const wrapper = shallow(<Counter />)
        const instance = wrapper.instance()
        instance.decrease()
        expect(instance.state).toMatchObject({
            count: {
                number: -1,
                modifyInfo: {
                    type: 'Subtraction'
                }
            }
        })

        instance.state.count.modifyInfo.step = 4
        instance.state.count.number = 8
        instance.decrease()
        expect(instance.state).toMatchObject({
            count: {
                number: 4,
                modifyInfo: {
                    type: 'Subtraction'
                }
            }
        })
    })
})
