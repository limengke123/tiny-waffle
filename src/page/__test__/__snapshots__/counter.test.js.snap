// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Count should render correctly 1`] = `
<div
  class="home"
>
  <div>
    <div
      class="count"
    >
      number:0
    </div>
    <div
      class="sub-info"
    >
      time:
    </div>
    <div
      class="sub-info"
    >
      type:
    </div>
    <div
      class="sub-info"
    >
      step:1
    </div>
  </div>
  <div>
    <button
      type="button"
    >
      -
    </button>
    <button
      type="button"
    >
      +
    </button>
  </div>
  <button
    type="button"
  >
    修改增长步长
  </button>
</div>
`;
